core/transientstore/store.go

```diff
@@ -80,11 +86,45 @@ type RwsetScanner struct {
 
 // NewStoreProvider instantiates TransientStoreProvider
 func NewStoreProvider(path string) (StoreProvider, error) {
-	dbProvider, err := leveldbhelper.NewProvider(&leveldbhelper.Conf{DBPath: path})
+	// Ensure the routine is invoked while the peer is down.
+	lockPath := filepath.Join(filepath.Dir(path), "fileLock")
```

Comment:
`transientStoreFileLock` would be a clearer name now - as this dir will directly be created inside `/var/production`, it may be hard to decipher by looking at the dir structure about the context of this dir.

Reply:
👍